src/Services/SearchService.php

```diff
@@ -117,10 +135,16 @@ public function handleSearchOptions($searchOptions, $request)
      */
     protected function search($request)
     {
+        if ($this->results) {
```

Comment:
I guess this was added to prevent the issue of double requests:
* This could cause caching issues in the future.
* It doesn't solve the actual problem. Plenty is still trying to make the same request twice.

It would be interesting why there are double requests since it didn't before (this story).

Reply:
The search function is called from two methods - handleSearchOptions and handleSearchQuery. Both of them are triggered on search request but previously I only worked with handleSearchQuery. I'm not sure what could have changed but for now I only commented the code which calls search function from handleSearchOptions method. All information is parsed and formatted inside handleSearchOptions so in the future maybe it could be possible to remove handleSearchOptions if it is not needed when implementing filters.